core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -2002,6 +2007,19 @@ def _update_model_attr(self, model: str, **attributes):
     def _log_model_stats(self, model, _is_refit=False):
         """Logs model fit time, val score, predict time, and predict_1_time"""
         model = self.load_model(model)
+        print_weights = model._get_tags().get("print_weights", False)
+
+        if print_weights:
+            model_weights = model._get_model_weights()
+            model_weights = {k: round(v, 3) for k, v in model_weights.items()}
+            msg_weights = ""
+            is_first = True
+            for key, value in sorted(model_weights.items(), key=lambda x: x[1], reverse=True):
+                if not is_first:
+                    msg_weights += ", "
+                msg_weights += f"'{key}': {value}"
+                is_first = False
+            logger.log(20, f"\tEnsemble Weights: {{{msg_weights}}}")
```

Comment:
Nit: Why do we need this logic for printing, would something like the following be enough?
```python
model_weights = model._get_model_weights()
model_weights = {k: round(v, 3) for k, v in sorted(model_weights.items(), key=lambda item: item[1], reverse=True)}
logger.log(20, f:\tEnsemble Weights: {str(model_weights)}")
```